tests/test_azure_power_off_vms.py

```diff
@@ -0,0 +1,316 @@
+"""Unit tests for reaper.azure_power_off_vms."""
+from unittest.mock import Mock, patch
+import uuid
+
+import reaper.azure_power_off_vms
+
+
+def synthesize_resource_id(
+    subscription_id=None,
+    resource_group=None,
+    resource_provider=None,
+    resource_type=None,
+    resource_id=None,
+):
+    """
+    Synthesize a pseudorandom resource ID.
+
+    Defaults assume this should look like a Compute Virtual Machine's ID.
+    """
+    return (
+        f"/subscriptions/{subscription_id if subscription_id else uuid.uuid4()}"
+        f"/resourceGroups/{resource_group if resource_group else uuid.uuid4()}"
+        f"/providers/{resource_provider if resource_provider else 'Microsoft.Compute'}"
+        f"/{resource_type if resource_type else 'virtualMachines'}"
+        f"/{resource_id if resource_id else uuid.uuid4()}"
+    )
+
+
+def synthesize_tags(with_bypass_tag=True):
+    """Synthesize a set of pseudorandom tags for a resource."""
+    conditional_tag_key = (
```

Comment:
is this used ?  I don't see it being returned.

Reply:
It's used as the key a few lines later, currently on line 38.

Reply:
Ahh never mind, you're right,  I was thinking  {  "conditional_tag_key": ... }  and not { conditional_tag_key: ... }